src/Documentation/Getting-Started-With-Orleans/Grain-LifeCycle.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: page
+title: Grain LifeCycle
+---
+
+
+## Grain LifeCycle
+
+We say grains virtually always exist but here we want to describe different stages of a grain's life.
+A grain is like a class definition, it is nothing unless it is on the memory and can execute code.
+An instance of a class is instantiated with the `new` keyword in C#.
+an instance of a grain is called an activation.
+The grain's instances are differentiated by their key and a grain with a specific key is either activated or not.
+You don't control this by using construction/destruction keywords and instead Orleans runtime manages this for you.
+
+All possible grains in the whole key space virtually always exist
+and it means whenever you call a method of a grain, if it is not active, it will get activated by the runtime.
+Also after the grain passes a certain amount of time in idle state, the runtime will deactivate the grain (i.e. destroys it).
+A grain is in idle state if it is not currently processing a message and its queue of messages is empty.
+
+When you get a grain reference in a client or in another grain by using `GetGrain`,
+you only get a reference to a grain and you don't get the activation.
+When you call a method using the reference,then the grain will be activated in a silo (if it is not already active)
+and your reference sends messages to the activation by calling methods and points to the activation logically.
+
+`GetGrain` itself is not asynchronous, can not fail and only returns the logical reference which points to the activation.
```

Comment:
Maybe "`GetGrain` itself is an inexpensive local operation of constructing a proxy object with an embedded identity of the target grain."?